### PR TITLE
admin controller endpoints for managing personas

### PR DESCRIPTION
**Pull request by ahsu1230**
@stkem @yingjieMiao

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala [hunk 1]
```diff
@@ -805,6 +806,11 @@ class AdminUserController @Inject() (
           keepRepo.getByUser(userId).foreach { bookmark => keepRepo.save(bookmark.withActive(false)) }
           collectionRepo.getUnfortunatelyIncompleteTagsByUser(userId).foreach { collection => collectionRepo.save(collection.copy(state = CollectionStates.INACTIVE)) }
 
+          // Libraries Data
+          libraryInviteRepo.getByUser(userId, Set(LibraryInviteStates.INACTIVE)).foreach { case (invite, _) => libraryInviteRepo.save(invite.withState(LibraryInviteStates.INACTIVE)) } // Library Invites
+          libraryMembershipRepo.getWithUserId(userId).foreach { membership => libraryMembershipRepo.save(membership.withState(LibraryMembershipStates.INACTIVE)) } // Library Memberships
+          libraryRepo.getAllByOwner(userId).foreach { library => libraryRepo.save(library.withState(LibraryStates.INACTIVE)) } // Libraries
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminUserController.scala:812]: when deactivating users, we should also delete library information. Currently, deactivated users still have active libraries & library memberships =/
@andrewconner @leogrim

**Comment by stkem** [comment]
lgtm

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPersonaController.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    request.body.asJson.map { json =>
+      val currentNameOpt = (json \ "name").asOpt[PersonaName]
+      val displayNameOpt = (json \ "displayName").asOpt[String]
+      val iconPathOpt = (json \ "iconPath").asOpt[String]
+      val activeIconPathOpt = (json \ "activeIconPath").asOpt[String]
+      val stateOpt = (json \ "state").asOpt[State[Persona]]
+
+      db.readWrite { implicit s =>
+        val currentPersona = personaRepo.get(id)
+        val newCurrentName = currentNameOpt.getOrElse(currentPersona.name)
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPersonaController.scala:75]: syntax-wise, these could just be 
`currentNameOpt orElse currentPersona.name`

slightly less crowded.

**Comment by yingjieMiao** [comment]
lgtm

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPersonaController.scala (hunk 2 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPersonaController.scala:75]: I think that will change the type of the return value that I want.
From my understanding, `orElse` requires an Option for the second parameter and returns an Option result. For this case, I'd like to just have `PersonaName` =D

**Comment by ahsu1230** [comment]
thanks for the reviews

**Pull request merged**